src/storage/link.cpp

```diff
@@ -10,19 +10,41 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "link.h"
+#include <QDebug>
 
-Link::Link(int linkId, const QString &urlString, const QString &thumbPath, const QString &title) :
-    m_linkId(linkId), m_url(urlString), m_thumbPath(thumbPath), m_title(title)
+Link::Link(int linkId, const QString &urlString, const QString &thumbPath, const QString &title, const QDateTime &date) :
+    m_linkId(linkId), m_url(urlString), m_thumbPath(thumbPath), m_title(title), m_date(date)
 {
+    QDateTime today = QDateTime::currentDateTime();
+    QDateTime yesterday = today.addDays(-1);
+    QString fullDateFormat = "dd.MM.yyyy";
+    QString fullDate = date.toString(fullDateFormat);
+
+    if (fullDate == today.toString(fullDateFormat)) {
+        //% "Today"
+        m_sectionName = qtTrId("sailfish_browser-se-today");
+    } else if (fullDate == yesterday.toString(fullDateFormat)) {
+        //% "Yesterday"
+        m_sectionName = qtTrId("sailfish_browser-se-yesterday");
+    } else {
+        QString standaloneMonthName = QLocale::system().standaloneMonthName(date.date().month());
+        QString year = QString::number(date.date().year());
+        m_sectionName = QString("%1 %2").arg(standaloneMonthName, year);
```

Comment:
This should be not fixed on the model side like rather platform style time formatting should be used.

Reply:
As noted in an another comment date itself is enough. Further note, translations on the C++ are not preferred. That said, this can be more or less dropped.

Reply:
qml side doesn't allow me to determine what the section will be

Reply:
I would check how it's done in the Messages app. Messages are grouped by date quite reasonably.

Reply:
I haven't found the option I needed in the message app. There, the role for the section property is grouped for 1 day. I need to group by month, and today and yesterday

Reply:
Right, grouping there has recently changed. Before, it was more like what you're doing and looked more reasonable to me. Oh well...

Reply:
`

        // Return the date for the past week, and group all older items together under an
        // arbitrary older date
        const int daysDiff = QDate::currentDate().toJulianDay() - dateTime.date().toJulianDay();
        if (daysDiff < 7)
            return dateTime.date();

        // Arbitrary static date for older items..
        return QDate(2000, 1, 1);
`
In this case, 8 sections are returned to the model - the first 7 days, and everything after. For a browser design, this option will not work, you need to group it like this: Today, Yesterday, September 2020, August 2020, etc...

Reply:
For my case, I don't see other way to store role for the section group